SmartDeviceLink/SDLChoiceSet.h

```diff
@@ -0,0 +1,140 @@
+//
+//  SDLChoiceSet.h
+//  SmartDeviceLink
+//
+//  Created by Joel Fischer on 5/21/18.
+//  Copyright © 2018 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@protocol SDLChoiceSetDelegate;
+@class SDLChoiceCell;
+@class SDLTTSChunk;
+@class SDLVRHelpItem;
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, SDLChoiceSetLayout) {
+    SDLChoiceSetLayoutList,
+    SDLChoiceSetLayoutTiles,
+};
+
+@interface SDLChoiceSet: NSObject
+
+/**
+ Set this to change the default timeout for all choice sets. If a timeout is not set on an individual choice set object (or if it is set to 0.0), then it will use this timeout instead. See `timeout` for more details. If this is not set by you, it will default to 10 seconds.
+ */
+@property (class, assign, nonatomic) NSTimeInterval defaultTimeout;
+
+/**
+ Set this to change the default layout for all choice sets. If a layout is not set on an individual choice set object, then it will use this layout instead. See `layout` for more details. If this is not set by you, it will default to `SDLChoiceSetLayoutList`.
+ */
+@property (class, assign, nonatomic) SDLChoiceSetLayout defaultLayout;
+
+/**
+ Maps to PerformInteraction.initialText. The title of the choice set, and/or the initial text on a keyboard prompt.
+
+ Required
+ */
+@property (copy, nonatomic) NSString *title;
+
+/**
+ Maps to PerformInteraction.initialPrompt. The initial prompt spoken to the user at the start of an interaction.
+
+ Optional
+ */
+@property (copy, nonatomic, nullable) NSArray<SDLTTSChunk *> *initialPrompt;
+
+/**
+ Maps to PerformInteraction.interactionLayout. Whether the presented choices are arranged as a set of tiles or a list.
+
+ Required
+ */
+@property (assign, nonatomic) SDLChoiceSetLayout layout;
+
+/**
+ Maps to PerformInteraction.timeout. This applies only to a manual selection (not a voice selection, which has its timeout handled by the system).
+
+ Required
+ */
+@property (assign, nonatomic) NSTimeInterval timeout;
+
+/**
+ Maps to PerformInteraction.timeoutPrompt. This text is spoken when a VR interaction times out. If this set is presented in a manual (non-voice) only interaction, this will be ignored.
+
+ Optional
+ */
+@property (copy, nonatomic, nullable) NSArray<SDLTTSChunk *> *timeoutPrompt;
+
+/**
+ Maps to PerformInteraction.helpPrompt. This is the spoken string when a user speaks "help" when the interaction is occurring.
+
+ Optional
+ */
+@property (copy, nonatomic, nullable) NSArray<SDLTTSChunk *> *helpPrompt;
+
+/**
+ Maps to PerformInteraction.vrHelp. This is a list of help text presented to the user when they are in a voice recognition interaction from your choice set of options. If this set is presented in a touch only interaction, this will be ignored.
+
+ Optional
+ */
+@property (copy, nonatomic, nullable) NSArray<SDLVRHelpItem *> *helpList;
```

Comment:
`SDLVRHelpItem`s have a required `position` parameter (starting with 1). It would be nice if the manager took care of assigning the positions. Also it would be nice if the manager handled uploading the artworks for the `SDLVRHelpItem`s.

Reply:
I set the position automatically, but we can’t upload artwork automatically, because it uses an SDLImage, not an SDLArtwork. We’d have to create a new class to mimic VRHelpItem but with an SDLArtwork and a string and no position. That wasn’t part of this proposal, but should be done eventually.